cloudinit/sources/DataSourceVMware.py

```diff
@@ -53,6 +55,20 @@
 WAIT_ON_NETWORK_IPV4 = "ipv4"
 WAIT_ON_NETWORK_IPV6 = "ipv6"
 
+SUPPORTED_UPDATE_EVENTS = {
+    # Support network reconfiguration for the following events:
+    # TODO(akutz) Add support for METADATA_CHANGE and USER_REQUEST when
+    #             those events are supported by Cloud-Init.
+    EventScope.NETWORK: {
+        EventType.BOOT,
+        EventType.BOOT_LEGACY,
+        EventType.BOOT_NEW_INSTANCE,
+        EventType.HOTPLUG,
+    }
+}
+
+SUPPORTED_UPDATE_EVENTS_GUEST_INFO_KEY = "cloudinit.supported-update-events"
```

Comment:
It seems we're overloading the word "supported" here. As currently used:
Supported event: An event that a datasource can react to if set as a default event OR enabled via user data
Default event: An event that will automatically trigger on the datasource without any additional user data
Allowed event: A supported event that is either part of the default events or a user specified event

It appears you're advertising allowed events, but calling it support events. Can we call them allowed events to not conflict with other cloud-init concepts/documentation?